client/components/pages/files/FilesTable.vue

```diff
@@ -26,53 +49,116 @@ provide(FileTableKey, {
   selectedRow,
   selectedRowId,
   updateSelectedRow,
-  removeSelectedRow,
   unselectRow,
+  refresh,
 })
 
 const contextMenu = ref<InstanceType<typeof PContextMenu> | null>(null)
 provide(FileTableContextMenuKey, contextMenu)
+
+function resetPage() {
+  page.value = DEFAULT_PAGE
+  pageSize.value = DEFAULT_PAGE_SIZE
+  sortDefinition.value = []
+  refresh()
+}
+
+function onPage(event: DataTablePageEvent) {
+  page.value = event.page
+  pageSize.value = event.rows
+}
+
+function onSort(event: DataTableSortEvent) {
+  if (event.multiSortMeta) {
+    sortDefinition.value = event.multiSortMeta?.map<SortObject>((sort: DataTableSortMeta) => ({
```

Comment:
cannot be undefined here since you garded the clause

```suggestion
    sortDefinition.value = event.multiSortMeta.map<SortObject>((sort: DataTableSortMeta) => ({
```